File: sanghyeok0908/프로그래머스/2/12941. 최솟값 만들기/최솟값 만들기.java
import java.util.*;

class Solution
{
    public int solution(int []A, int []B)
    {
        int N = A.length;
        int answer = 0;
        Arrays.sort(A);
        Arrays.sort(B);
        
        for (int i = 0; i < N; i++) {
            answer += A[i] * B[N - 1 - i];
        }

        return answer;
    }
}

File: sanghyeok0908/프로그래머스/2/12941. 최솟값 만들기/최솟값 만들기.js
function solution(A,B){
    const n = A.length;
    let a = 0, b = 0;
    
    A.sort((o1, o2) => o1 - o2);
    B.sort((o1, o2) => o1 - o2);
    
    for (let i = 0; i < n; i++) {
        a += A[i] * B[n - 1 - i];
        b += A[n - 1 - i] * B[i];
        // console.log(a + " " + b);
    }
    return a > b ? b : a;
}